src/service.rs

```diff
@@ -108,18 +116,14 @@ pub struct ReceivedTransaction<Types: NodeType> {
 #[derive(Debug)]
 pub struct GlobalState<Types: NodeType> {
     // data store for the blocks
-    pub block_hash_to_block: HashMap<(BuilderCommitment, Types::Time), BlockInfo<Types>>,
+    pub blocks: lru::LruCache<BlockId<Types>, BlockInfo<Types>>,
```

Comment:
Am I correctly understanding that we replace `HashMap` with `LruCache` here to retrieve blocks ordered by visit frequencies? E.g., we call `promote` in `update_global_state` which wasn't supported before.

However, I don't quite understand why we need `LruCache` for this purpose. If we want to prevent returning duplicate blocks, can't we just call `remove` on a `HashMap`? Maybe I'm missing some more complicated use cases.

Reply:
It's not really used as a cache here. The reason I've replaced it is that there wasn't any sort of GC process for built blocks, so I'm just using it to forget old blocks, as it has a fixed capacity and pops least used blocks when full.
We can't just remove blocks when they are claimed, because there's no guarantee a block will be claimed at all.